### ISSUE TITLE
Optional Form Button Positioning

### ISSUE THREAD

**Issue by james-burkill** [Sage/carbon#6516]
### Desired behaviour

I have an accessibility requirement to position buttons on a form in such a way that the save button appears on the right-hand side of the form, while the cancel button appears on the left.  We would like more options for controlling where the buttons are positioned. 

### Current behaviour

Currently the Form component allows users to specify a saveButton, leftSideButtons and rightSideButtons and, although it is possible to set the alignment of buttons via the buttonAllignment attribute, the buttons are always displayed together.

### Suggested Solution

Either a new set of attributes allowing us to set different alignments for each button group, or a new buttonAlignment option such as justifiedLeftAndRight, based on the typographic alignment with the same name.  In the event of there being only leftSideButtons and a saveButton, this would place the leftSideButtons on the left and saveButton on the right.  In the event of there being only a saveButton and rightSideButtons, it could put the save button on the left and rightSideButtons on the right, and in the event of all three being specified, it might put the save button in the center, the leftSideButtons on the left and rightSideButtons on the right. Ultimately, this request is only motiviated by the need for the first of the three options.

### CodeSandbox or Storybook URL

SKPU-10765

### Anything else we should know?

The request comes from SBC Payroll UKI.

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

**Comment by edleeks87** [comment]
Notes for triage: all button props in Form accept node so it is already possible to customise the button layout, so focus should be on if this is what we advise or if we need to add the enhancement

![image](https://github.com/Sage/carbon/assets/44157880/8b78d0bd-51a2-4af8-ae75-63b03c1f5169)

**Comment by sianford** [comment]
@harpalsingh do you think this is something that should be added as an additional feature? Ed has demonstrated that the button positioning can be amended using a flex box, however he clarifies that then wouldn't render the Form Summary so that would need exporting along with the flex box method for swapping the button positions.

**Comment by Parsium** [comment]
Hey @harpalsingh, do you have any thoughts on Ed's suggestion?

**Comment by Parsium** [comment]
Hi @james-burkill 👋🏼 Would the solution Ed has described help you achieve the layout you'll like in the meantime?

**Comment by Parsium** [comment]
Hi @james-burkill, have you had a chance to look at @edleeks87's proposed layout?

**Comment by james-burkill** [comment]
Yes, this does the job.  We're not using form summary functionality.

**Comment by Parsium** [comment]
Thanks for the update @james-burkill 👍🏼 In that case, I'll close this issue
